File: November2021/HammingDistance.java
class Solution {
	// TC : O(1)
	// SC : O(1)
	public int hammingDistance(int x, int y) {
		int xor = x^y;
		int setBitCount = 0;
		for(int i=0;i<32;i++){ // 32 bit
			if(((xor>>i)&1)==1){ // right
				setBitCount++;
			}
		}
		return setBitCount;
	}
}

File: November2021/C++/Hamming_Distance.cpp
// author :shyam2520
// https://leetcode.com/problems/hamming-distance/

class Solution {
public:
    int hammingDistance(int x, int y) {
        x^=y;
        int count=0;
        while(x)
        {
            x&=x-1;
            count++;
        }
        return count;

    }
};